### PR TITLE
Add ewatercycle.observation.caravan.get_caravan_data()

### LINKED ISSUE #432: Caravan observations
The Caravan dataset also contains streamflow and ewatercycle has a observations module for grdc and usgs.  I woudl be nice if there was a `ewatercycle.observations.caravan.get_caravan_data(basin_id, ...)` method.

A bonus would be if it could have meta data regarding the gauge/station location.

With existing CaravanForcing I can get streamflow with

```python
from pathlib import Path
from rich import print
from ewatercycle.forcing import sources

path = Path.cwd()
forcing_path = path / "Forcing"/ "Camels3"
forcing_path.mkdir(parents=True, exist_ok=True)
experiment_start_date = "1997-08-01T00:00:00Z"
experiment_end_date = "2005-09-01T00:00:00Z"
HRU_id = 1022500

camels_forcing = sources['CaravanForcing'].generate(
                                start_time = experiment_start_date,
                                end_time = experiment_end_date,
                                directory = forcing_path ,
                                basin_id = f"camels_0{HRU_id}",
                                variables=('timezone', 'name', 'country', 'lat', 'lon', 'area', "streamflow")
                                                        )
print(camels_forcing)
```

### PR DESCRIPTION
**Pull request by sverhoeven**
Fixes #432

### CODE REVIEW

File: src/ewatercycle/observation/caravan.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    (see https://doi.org/10.1038/s41597-023-01975-w).
+
+    This retrieves it from the OpenDAP server of 4TU,
+    (see https://doi.org/10.4121/bf0eaf7c-f2fa-46f6-b8cd-77ad939dd350.v4).
+
+    Parameters:
+        basin_id: The ID of the desired basin. Data sets can be explored using
+                :py:func:`ewatercycle._forcings.caravan.CaravanForcing.get_dataset`
+                or :py:func:`ewatercycle._forcings.caravan.CaravanForcing.get_basin_id`
+                For more information do `help(CaravanForcing.get_basin_id)` or see
```
Comment (sverhoeven) [inline src/ewatercycle/observation/caravan.py:25]: Links dont show see https://github.com/eWaterCycle/ewatercycle/issues/436

**Review by BSchilperoort — APPROVED** [review]
Example code worked. Tried a different basin in Czechia, worked as well :+1: 

Nice to have an additional source of observations.

File: docs/observations.rst [hunk 2]
```diff
@@ -13,3 +13,9 @@ GRDC
 ----
 
 The `Global Runoff Data Centre <https://grdc.bafg.de/>`_ provides discharge data for a large number of stations around the world. In eWaterCycle we support GRDC data. This is not downloaded automatically, but required to be present on the infrastructure where the eWaterCycle platform is deployed. By special permission from GRDC our own instance contains data from the ArcticHYCOS and GCOS/GTN-H, GTN-R projects.
+
+Caravan
+-------
+
+The `Caravan <https://doi.org/10.1038/s41597-023-01975-w>`_ dataset contains river discharge for each of its basins.
+Observations can be retrieved using the :py:func:`ewatercycle.observation.caravan.get_caravan_data` function.
```
Comment (BSchilperoort) [inline docs/observations.rst:21]: Could be nice to link to https://www.ewatercycle.org/caravan-map/ so it's easy to find basins.

**Comment by sverhoeven** [comment]
Thanks for reviewing

**Pull request merged**